meilisearch-http/src/routes/indexes/settings.rs

```diff
@@ -165,7 +165,49 @@ make_setting_route!(
     "/typo-tolerance",
     meilisearch_lib::index::updates::TypoSettings,
     typo_tolerance,
-    "typoTolerance"
+    "typoTolerance",
+    analytics,
+    |setting: &Option<meilisearch_lib::index::updates::TypoSettings>, req: &HttpRequest| {
+        use serde_json::json;
+
+        analytics.publish(
+            "TypoToleranceUpdated Updated".to_string(),
+            json!({
+                "typo_tolerance": {
+                    "enabled": setting.as_ref().map(|s| !matches!(s.enabled, Setting::Set(false))).unwrap_or(true),
+                    "disable_on_attributes": setting
+                        .as_ref()
+                        .map(|s| s.disable_on_attributes.as_ref().set().map(|m| !m.is_empty()))
+                        .flatten()
+                        .unwrap_or(false),
+                    "disable_on_words": setting
+                        .as_ref()
+                        .map(|s| s.disable_on_words.as_ref().set().map(|m| !m.is_empty()))
+                        .flatten()
+                        .unwrap_or(false),
+                    "min_word_size_for_one_typo": setting
+                        .as_ref()
+                        .map(|s| s.min_word_size_for_typos
+                            .as_ref()
+                            .set()
+                            .map(|s| s.one_typo.set()))
+                        .flatten()
+                        .flatten()
+                        .unwrap_or(5),
+                    "min_word_size_for_two_typos": setting
+                        .as_ref()
+                        .map(|s| s.min_word_size_for_typos
+                            .as_ref()
+                            .set()
+                            .map(|s| s.two_typos.set()))
+                        .flatten()
+                        .flatten()
+                        .unwrap_or(9),
```

Comment:
Just out of curiosity, is it a good thing to specify constants here too? e.g. `.unwrap_or(9)`. If one day these values change, we should not forget them here I guess?

Reply:
Yeah, it could be a good idea to create two constants to hold these values so everything always stays synchronized.